lib/kitchen/driver/ssh_base.rb

```diff
@@ -115,6 +115,14 @@ def verify(state) # rubocop:disable Metrics/AbcSize
       rescue Kitchen::Transport::TransportFailed => ex
         raise ActionFailed, ex.message
       ensure
+        if verifier[:scp_file]
```

Comment:
Why did you decide to put this in the `ensure` block?  I would think you would want it to go in the existing connection block.